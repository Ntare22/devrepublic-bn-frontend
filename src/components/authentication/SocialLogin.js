import React from 'react';
import PropTypes from 'prop-types';
import googlesvg from '../../assets/google.svg';
import facebookSvg from '../../assets/facebook.svg';
import '../../scss/socialLogin.scss';

const SocialLogin = ({ name }) => (
  <a className="social-login-link" href={`${process.env.BACKEND_LINK}/api/v1/auth/${name}`}>
    <img src={name === 'google' ? googlesvg : facebookSvg} alt={name === 'google' ? 'google' : 'facebook'} />
  </a>
);

SocialLogin.propTypes = {
  name: PropTypes.oneOf(['facebook', 'google']).isRequired,
};
export default SocialLogin;
