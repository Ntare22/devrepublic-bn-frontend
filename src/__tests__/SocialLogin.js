import React from 'react';
import { shallow } from 'enzyme';
import SocialLogin from '../components/authentication/SocialLogin';

const testLogin = (name) => {
  const wrapper = shallow(<SocialLogin name={name} />);
  const result = wrapper.props();
  expect(result.href).toEqual(expect.stringContaining(`/api/v1/auth/${name}`));
};
describe('social login componet', () => {
  it('renders correctly', () => {
    const Component = <SocialLogin name="google" />;
    expect(Component).toMatchSnapshot();
  });
  it('should have a link element with google in the href', () => {
    testLogin('google');
  });
  it('should have a link element with facebook in the href', () => {
    testLogin('facebook');
  });
});
