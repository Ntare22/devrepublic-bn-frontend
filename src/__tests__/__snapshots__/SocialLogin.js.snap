// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`social login componet renders correctly 1`] = `
<SocialLogin
  name="google"
/>
`;
